MatrixSDK/Background/MXBackgroundSyncService.swift

```diff
@@ -500,5 +500,11 @@ public enum MXBackgroundSyncServiceError: Error {
             MXLog.debug("[MXBackgroundSyncService] updateBackgroundServiceStoresIfNeeded: Reset MXBackgroundCryptoStore")
             crypto.reset()
         }
+        
+        if let accountData = syncResponseStoreManager.syncResponseStore.accountData {
+            pushRulesManager.handleAccountData(accountData)
+        } else if let accountData = store.userAccountData ?? nil {
+            pushRulesManager.handleAccountData(accountData)
+        }
```

Comment:
This is copy and paste from the init.
Please create a new method like this `func updatePushRules` and call it from both.

Reply:
Actually this doesn't look like the right spot to put this code.
The method where you are adding this code in is called `func updateBackgroundServiceStoresIfNeeded()` which isn't what we are doing here
.
Probably a better spot would be here:
```swift
    private func fetchAndProcessEvent(withEventId eventId: String, roomId: String) {
        MXLog.debug("[NotificationService] fetchAndProcessEvent")
                
        NotificationService.backgroundSyncService.event(withEventId: eventId, inRoom: roomId) { [weak self] (response) in
            switch response {
            case .success(let event):
                MXLog.debug("[NotificationService] fetchAndProcessEvent: Event fetched successfully")
                self?.checkPlaybackAndContinueProcessing(event, roomId: roomId)
            case .failure(let error):
                MXLog.error("[NotificationService] fetchAndProcessEvent: Failed fetching notification event", context: error)
                self?.fallbackToBestAttemptContent(forEventId: eventId)
            }
        }
    }
```
in the `.success` case just before `checkPlaybackAndContinueProcessing(event, roomId: roomId)`
Wdyt?